<?xml version="1.0" encoding="UTF-8"?>

<launch>
<!--<arg name="des_x" value="0" />
    <arg name="des_y" value="0" /> 
    <group ns="/bat_swarm_red">
        <param name="Qmax" value="0.20"/>
        <param name="Qmin" value="0.06"/>
        <param name="NP" value="6"/>
        <param name="N_Gen" value="10000"/>
        <param name="D" value="2"/>
        <param name="Lower" value="-10"/>
        <param name="Upper" value="10"/>
        <param name="alpha" value="0.1"/>
        <param name="gamma" value="0.05"/>
        <node pkg="bat_algo" type="best_bat.py" name="best_bat" output="screen" />
        <node pkg="bat_algo" type="gen_checker.py" name="gen_checker" output = "screen" />
        <group ns="bat_1">
            <param name="initial_x" value="-8" />
            <param name="initial_y" value="-7" />
            <param name="des_pos_x" value="$(eval arg ('des_x')-3)" />
            <param name="des_pos_y" value="$(eval arg ('des_y')+3)" />
            <param name="des_ori_x" value="0" />
            <param name="des_ori_y" value="0" />
            <param name="des_ori_z" value="0" />
            <param name="des_ori_w" value="1" />
            <param name="A" value="0.69"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_1"/>
            <param name="fix_orientation" value="0" />
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_1" output="screen" />
        </group>
        <group ns="bat_2">
            <param name="initial_x" value="-8" />
            <param name="initial_y" value="-4" />
            <param name="des_pos_x" value="$(eval arg ('des_x')-1.5)" />
            <param name="des_pos_y" value="$(eval arg ('des_y')+1.5)" />
            <param name="des_ori_x" value="0" />
            <param name="des_ori_y" value="0" />
            <param name="des_ori_z" value="0" />
            <param name="des_ori_w" value="1" />
            <param name="A" value="0.85"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_2"/>
            <param name="fix_orientation" value="0" />
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_2" output="screen" />
        </group>
        <group ns="bat_3">
            <param name="initial_x" value="-8.0" />
            <param name="initial_y" value="-1.0" />
            <param name="des_pos_x" value="$(arg des_x)" />
            <param name="des_pos_y" value="$(arg des_y)" />
            <param name="des_ori_x" value="0" />
            <param name="des_ori_y" value="0" />
            <param name="des_ori_z" value="0" />
            <param name="des_ori_w" value="1" />
            <param name="A" value="0.78"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_3"/>
            <param name="fix_orientation" value="0" />
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_3" output="screen" />
        </group>
        <group ns="bat_4">
            <param name="initial_x" value="-8.0" />
            <param name="initial_y" value="2.0" />
            <param name="des_pos_x" value="$(eval arg ('des_x')-1.5)" />
            <param name="des_pos_y" value="$(eval arg ('des_y')-1.5)" />
            <param name="des_ori_x" value="0" />
            <param name="des_ori_y" value="0" />
            <param name="des_ori_z" value="0" />
            <param name="des_ori_w" value="1" />
            <param name="A" value="0.89"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_4"/>
            <param name="fix_orientation" value="0" />
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_4" output="screen" />
        </group>
        <group ns="bat_5">
            <param name="initial_x" value="-8.0" />
            <param name="initial_y" value="5.0" />
            <param name="des_pos_x" value="$(eval arg ('des_x')-3)" />
            <param name="des_pos_y" value="$(eval arg ('des_y')-3)" />
            <param name="des_ori_x" value="0" />
            <param name="des_ori_y" value="0" />
            <param name="des_ori_z" value="0" />
            <param name="des_ori_w" value="1" />
            <param name="A" value="0.78"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_5"/>
            <param name="fix_orientation" value="0" />
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_5" output="screen" />
        </group>
    </group>
    <group ns="/bat_swarm_green">
        <param name="Qmax" value="0.20"/>
        <param name="Qmin" value="0.06"/>
        <param name="NP" value="6"/>
        <param name="N_Gen" value="10000"/>
        <param name="D" value="2"/>
        <param name="Lower" value="-10"/>
        <param name="Upper" value="10"/>
        <param name="alpha" value="0.1"/>
        <param name="gamma" value="0.05"/>
        <node pkg="bat_algo" type="best_bat.py" name="best_bat" output="screen" />
        <node pkg="bat_algo" type="gen_checker.py" name="gen_checker" output = "screen" />
        <group ns="bat_6">
            <param name="initial_x" value="-8" />
            <param name="initial_y" value="-6" />
            <param name="des_pos_x" value="$(eval arg ('des_x')-3)" />
            <param name="des_pos_y" value="$(eval arg ('des_y')+3)" />
            <param name="des_ori_x" value="0" />
            <param name="des_ori_y" value="0" />
            <param name="des_ori_z" value="0" />
            <param name="des_ori_w" value="1" />
            <param name="A" value="0.69"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_6"/>
            <param name="fix_orientation" value="0" />
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_1" output="screen" />
        </group>
        <group ns="bat_7">
            <param name="initial_x" value="-8" />
            <param name="initial_y" value="-3" />
            <param name="des_pos_x" value="$(eval arg ('des_x')-1.5)" />
            <param name="des_pos_y" value="$(eval arg ('des_y')+1.5)" />
            <param name="des_ori_x" value="0" />
            <param name="des_ori_y" value="0" />
            <param name="des_ori_z" value="0" />
            <param name="des_ori_w" value="1" />
            <param name="A" value="0.85"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_7"/>
            <param name="fix_orientation" value="0" />
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_2" output="screen" />
        </group>
        <group ns="bat_8">
            <param name="initial_x" value="-8.0" />
            <param name="initial_y" value="3.0" />
            <param name="des_pos_x" value="$(arg des_x)" />
            <param name="des_pos_y" value="$(arg des_y)" />
            <param name="des_ori_x" value="0" />
            <param name="des_ori_y" value="0" />
            <param name="des_ori_z" value="0" />
            <param name="des_ori_w" value="1" />
            <param name="A" value="0.78"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_8"/>
            <param name="fix_orientation" value="0" />
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_3" output="screen" />
        </group>
        <group ns="bat_9">
            <param name="initial_x" value="-8.0" />
            <param name="initial_y" value="0.0" />
            <param name="des_pos_x" value="$(eval arg ('des_x')-1.5)" />
            <param name="des_pos_y" value="$(eval arg ('des_y')-1.5)" />
            <param name="des_ori_x" value="0" />
            <param name="des_ori_y" value="0" />
            <param name="des_ori_z" value="0" />
            <param name="des_ori_w" value="1" />
            <param name="A" value="0.89"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_9"/>
            <param name="fix_orientation" value="0" />
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_4" output="screen" />
        </group>
        <group ns="bat_10">
            <param name="initial_x" value="-8.0" />
            <param name="initial_y" value="6.0" />
            <param name="des_pos_x" value="$(eval arg ('des_x')-3)" />
            <param name="des_pos_y" value="$(eval arg ('des_y')-3)" />
            <param name="des_ori_x" value="0" />
            <param name="des_ori_y" value="0" />
            <param name="des_ori_z" value="0" />
            <param name="des_ori_w" value="1" />
            <param name="A" value="0.78"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_10"/>
            <param name="fix_orientation" value="0" />
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_5" output="screen" />
        </group>
    </group>
    <group ns="/bat_swarm_blue">
        <param name="Qmax" value="0.20"/>
        <param name="Qmin" value="0.06"/>
        <param name="NP" value="6"/>
        <param name="N_Gen" value="10000"/>
        <param name="D" value="2"/>
        <param name="Lower" value="-10"/>
        <param name="Upper" value="10"/>
        <param name="alpha" value="0.1"/>
        <param name="gamma" value="0.05"/>
        <node pkg="bat_algo" type="best_bat.py" name="best_bat" output="screen" />
        <node pkg="bat_algo" type="gen_checker.py" name="gen_checker" output = "screen" />
        <group ns="bat_11">
            <param name="initial_x" value="-8" />
            <param name="initial_y" value="-5" />
            <param name="des_pos_x" value="$(eval arg ('des_x')-3)" />
            <param name="des_pos_y" value="$(eval arg ('des_y')+3)" />
            <param name="des_ori_x" value="0" />
            <param name="des_ori_y" value="0" />
            <param name="des_ori_z" value="0" />
            <param name="des_ori_w" value="1" />
            <param name="A" value="0.69"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_11"/>
            <param name="fix_orientation" value="0" />
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_1" output="screen" />
        </group>
        <group ns="bat_12">
            <param name="initial_x" value="-8" />
            <param name="initial_y" value="-2" />
            <param name="des_pos_x" value="$(eval arg ('des_x')-1.5)" />
            <param name="des_pos_y" value="$(eval arg ('des_y')+1.5)" />
            <param name="des_ori_x" value="0" />
            <param name="des_ori_y" value="0" />
            <param name="des_ori_z" value="0" />
            <param name="des_ori_w" value="1" />
            <param name="A" value="0.85"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_12"/>
            <param name="fix_orientation" value="0" />
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_2" output="screen" />
        </group>
        <group ns="bat_13">
            <param name="initial_x" value="-8.0" />
            <param name="initial_y" value="1.0" />
            <param name="des_pos_x" value="$(arg des_x)" />
            <param name="des_pos_y" value="$(arg des_y)" />
            <param name="des_ori_x" value="0" />
            <param name="des_ori_y" value="0" />
            <param name="des_ori_z" value="0" />
            <param name="des_ori_w" value="1" />
            <param name="A" value="0.78"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_13"/>
            <param name="fix_orientation" value="0" />
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_3" output="screen" />
        </group>
        <group ns="bat_14">
            <param name="initial_x" value="-8.0" />
            <param name="initial_y" value="4.0" />
            <param name="des_pos_x" value="$(eval arg ('des_x')-1.5)" />
            <param name="des_pos_y" value="$(eval arg ('des_y')-1.5)" />
            <param name="des_ori_x" value="0" />
            <param name="des_ori_y" value="0" />
            <param name="des_ori_z" value="0" />
            <param name="A" value="0.89"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_14"/>
            <param name="fix_orientation" value="0" />
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_4" output="screen" />
        </group>
        <group ns="bat_15">
            <param name="initial_x" value="-8.0" />
            <param name="initial_y" value="7.0" />
            <param name="des_pos_x" value="$(eval arg ('des_x')-3)" />
            <param name="des_pos_y" value="$(eval arg ('des_y')-3)" />
            <param name="des_ori_x" value="0" />
            <param name="des_ori_y" value="0" />
            <param name="des_ori_z" value="0" />
            <param name="des_ori_w" value="1" />
            <param name="A" value="0.78"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_15"/>
            <param name="fix_orientation" value="0" />
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_5" output="screen" />
        </group>
    </group> -->
    <group ns="/bat_swarm">
        <arg name="des_x" value="0" />
        <arg name="des_y" value="0" /> 
        <param name="Qmax" value="0.20"/>
        <param name="Qmin" value="0.06"/>
        <param name="NP" value="6"/>
        <param name="N_Gen" value="10000"/>
        <param name="D" value="2"/>
        <param name="Lower" value="-10"/>
        <param name="Upper" value="10"/>
        <param name="alpha" value="0.1"/>
        <param name="gamma" value="0.05"/>
        <node pkg="bat_algo" type="best_bat.py" name="best_bat" output="screen" />
        <node pkg="bat_algo" type="gen_checker.py" name="gen_checker" output = "screen" />
        <group ns="bat_1">
            <param name="initial_x" value="-8" />
            <param name="initial_y" value="-7" />
            <param name="des_pos_x" value="$(eval arg ('des_x')-4-3)" />
            <param name="des_pos_y" value="$(eval arg ('des_y')+7+3)" />
            <param name="des_ori_x" value="0" />
            <param name="des_ori_y" value="0" />
            <param name="des_ori_z" value="0" />
            <param name="des_ori_w" value="1" />
            <param name="A" value="0.69"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_1"/>
            <param name="fix_orientation" value="0" />
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_1" output="screen" />
        </group>
        <group ns="bat_2">
            <param name="initial_x" value="-8" />
            <param name="initial_y" value="-4" />
            <param name="des_pos_x" value="$(eval arg ('des_x')-4-1.5)" />
            <param name="des_pos_y" value="$(eval arg ('des_y')+7+1.5)" />
            <param name="des_ori_x" value="0" />
            <param name="des_ori_y" value="0" />
            <param name="des_ori_z" value="0" />
            <param name="des_ori_w" value="1" />
            <param name="A" value="0.85"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_2"/>
            <param name="fix_orientation" value="0" />
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_2" output="screen" />
        </group>
        <group ns="bat_3">
            <param name="initial_x" value="-8.0" />
            <param name="initial_y" value="-1.0" />
            <param name="des_pos_x" value="$(arg des_x)-4" />
            <param name="des_pos_y" value="$(arg des_y)+7" />
            <param name="des_ori_x" value="0" />
            <param name="des_ori_y" value="0" />
            <param name="des_ori_z" value="0" />
            <param name="des_ori_w" value="1" />
            <param name="A" value="0.78"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_3"/>
            <param name="fix_orientation" value="0" />
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_3" output="screen" />
        </group>
        <group ns="bat_4">bat_swarm_blue
            <param name="initial_x" value="-8.0" />
            <param name="initial_y" value="2.0" />
            <param name="des_pos_x" value="$(eval arg ('des_x')-4-1.5)" />
            <param name="des_pos_y" value="$(eval arg ('des_y')+7-1.5)" />
            <param name="des_ori_x" value="0" />
            <param name="des_ori_y" value="0" />
            <param name="des_ori_z" value="0" />
            <param name="des_ori_w" value="1" />
            <param name="A" value="0.89"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_4"/>
            <param name="fix_orientation" value="0" />
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_4" output="screen" />
        </group>
        <group ns="bat_5">
            <param name="initial_x" value="-8.0" />
            <param name="initial_y" value="5.0" />
            <param name="des_pos_x" value="$(eval arg ('des_x')-4-3)" />
            <param name="des_pos_y" value="$(eval arg ('des_y')+7-3)" />
            <param name="des_ori_x" value="0" />
            <param name="des_ori_y" value="0" />
            <param name="des_ori_z" value="0" />
            <param name="des_ori_w" value="1" />
            <param name="A" value="0.78"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_5"/>
            <param name="fix_orientation" value="0" />
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_5" output="screen" />
        </group>
        <group ns="bat_6">
            <param name="initial_x" value="-8" />
            <param name="initial_y" value="-6" />
            <param name="des_pos_x" value="$(eval arg ('des_x')-3)" />
            <param name="des_pos_y" value="$(eval arg ('des_y')+3)" />
            <param name="des_ori_x" value="0" />
            <param name="des_ori_y" value="0" />
            <param name="des_ori_z" value="0" />
            <param name="des_ori_w" value="1" />
            <param name="A" value="0.69"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_6"/>
            <param name="fix_orientation" value="0" />
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_1" output="screen" />
        </group>
        <group ns="bat_7">
            <param name="initial_x" value="-8" />
            <param name="initial_y" value="-3" />
            <param name="des_pos_x" value="$(eval arg ('des_x')-1.5)" />
            <param name="des_pos_y" value="$(eval arg ('des_y')+1.5)" />
            <param name="des_ori_x" value="0" />
            <param name="des_ori_y" value="0" />
            <param name="des_ori_z" value="0" />
            <param name="des_ori_w" value="1" />
            <param name="A" value="0.85"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_7"/>
            <param name="fix_orientation" value="0" />
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_2" output="screen" />
        </group>
        <group ns="bat_8">
            <param name="initial_x" value="-8.0" />
            <param name="initial_y" value="3.0" />
            <param name="des_pos_x" value="$(arg des_x)" />
            <param name="des_pos_y" value="$(arg des_y)" />
            <param name="des_ori_x" value="0" />
            <param name="des_ori_y" value="0" />
            <param name="des_ori_z" value="0" />
            <param name="des_ori_w" value="1" />
            <param name="A" value="0.78"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_8"/>
            <param name="fix_orientation" value="0" />
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_3" output="screen" />
        </group>
        <group ns="bat_9">
            <param name="initial_x" value="-8.0" />
            <param name="initial_y" value="0.0" />
            <param name="des_pos_x" value="$(eval arg ('des_x')-1.5)" />
            <param name="des_pos_y" value="$(eval arg ('des_y')-1.5)" />
            <param name="des_ori_x" value="0" />
            <param name="des_ori_y" value="0" />
            <param name="des_ori_z" value="0" />
            <param name="des_ori_w" value="1" />
            <param name="A" value="0.89"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_9"/>
            <param name="fix_orientation" value="0" />
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_4" output="screen" />
        </group>
        <group ns="bat_10">
            <param name="initial_x" value="-8.0" />
            <param name="initial_y" value="6.0" />
            <param name="des_pos_x" value="$(eval arg ('des_x')-3)" />
            <param name="des_pos_y" value="$(eval arg ('des_y')-3)" />
            <param name="des_ori_x" value="0" />
            <param name="des_ori_y" value="0" />
            <param name="des_ori_z" value="0" />
            <param name="des_ori_w" value="1" />
            <param name="A" value="0.78"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_10"/>
            <param name="fix_orientation" value="0" />
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_5" output="screen" />
        </group>
        <group ns="bat_11">
            <param name="initial_x" value="-8" />
            <param name="initial_y" value="-5" />
            <param name="des_pos_x" value="$(eval arg ('des_x')-4-3)" />
            <param name="des_pos_y" value="$(eval arg ('des_y')-7+3)" />
            <param name="des_ori_x" value="0" />
            <param name="des_ori_y" value="0" />
            <param name="des_ori_z" value="0" />
            <param name="des_ori_w" value="1" />
            <param name="A" value="0.69"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_11"/>
            <param name="fix_orientation" value="0" />
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_1" output="screen" />
        </group>
        <group ns="bat_12">
            <param name="initial_x" value="-8" />
            <param name="initial_y" value="-2" />
            <param name="des_pos_x" value="$(eval arg ('des_x')-4-1.5)" />
            <param name="des_pos_y" value="$(eval arg ('des_y')-7+1.5)" />
            <param name="des_ori_x" value="0" />
            <param name="des_ori_y" value="0" />
            <param name="des_ori_z" value="0" />
            <param name="des_ori_w" value="1" />
            <param name="A" value="0.85"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_12"/>
            <param name="fix_orientation" value="0" />
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_2" output="screen" />
        </group>
        <group ns="bat_13">
            <param name="initial_x" value="-8.0" />
            <param name="initial_y" value="1.0" />
            <param name="des_pos_x" value="$(arg des_x)-4" />
            <param name="des_pos_y" value="$(arg des_y)-7" />
            <param name="des_ori_x" value="0" />
            <param name="des_ori_y" value="0" />
            <param name="des_ori_z" value="0" />
            <param name="des_ori_w" value="1" />
            <param name="A" value="0.78"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_13"/>
            <param name="fix_orientation" value="0" />
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_3" output="screen" />
        </group>
        <group ns="bat_14">
            <param name="initial_x" value="-8.0" />
            <param name="initial_y" value="4.0" />
            <param name="des_pos_x" value="$(eval arg ('des_x')-4-1.5)" />
            <param name="des_pos_y" value="$(eval arg ('des_y')-7-1.5)" />
            <param name="des_ori_x" value="0" />
            <param name="des_ori_y" value="0" />
            <param name="des_ori_z" value="0" />
            <param name="A" value="0.89"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_14"/>
            <param name="fix_orientation" value="0" />
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_4" output="screen" />
        </group>
        <group ns="bat_15">
            <param name="initial_x" value="-8.0" />
            <param name="initial_y" value="7.0" />
            <param name="des_pos_x" value="$(eval arg ('des_x')-4-3)" />
            <param name="des_pos_y" value="$(eval arg ('des_y')-7-3)" />
            <param name="des_ori_x" value="0" />
            <param name="des_ori_y" value="0" />
            <param name="des_ori_z" value="0" />
            <param name="des_ori_w" value="1" />
            <param name="A" value="0.78"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_15"/>
            <param name="fix_orientation" value="0" />
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_5" output="screen" />
        </group>
    </group>
</launch>